### ISSUE TITLE
Egregious Code Duplication in PowerFlows

### ISSUE THREAD

**Issue by GabrielKS** [NREL-Sienna/PowerFlows.jl#43]
The block:
```julia
    for (bus_no, ix) in bus_lookup
        bus_name = temp_bus_map[bus_no]
        bus = PSY.get_component(PSY.Bus, sys, bus_name)
        bus_type[ix] = PSY.get_bustype(bus)
        if bus_type[ix] == PSY.ACBusTypes.REF
            bus_angles[ix] = 0.0
        else
            bus_angles[ix] = PSY.get_angle(bus)
        end
        bus_magnitude[ix] = PSY.get_magnitude(bus)
    end
```

appears four times in `PowerFlowData.jl`. I'm sure there is more of this.

**Comment by jd-lara** [comment]
This might have been me.

**Comment by rodrigomha** [comment]
Closed via #45
